tests/test_metadata_harvester_cli.py

```diff
@@ -28,75 +10,239 @@ def kielipankki_api_url():
+        "pid2": {"infor": "mation", "id": "pid2"},
     }
-               }
+    metadata_harvester_cli.send_data_to_metax(record_dict)
+
+    assert mock_requests_post.call_count == 4
+
+    post_requests = mock_requests_post.request_history[1::2]
+    for post_request, record in zip(post_requests, record_dict.values()):
+        assert post_request.method == "POST"
+        assert post_request.json()["id"] == record["id"]
 
-#These tests hang when get_all_metadata_records() does not have a limit. When changing value to "1", the tests pass without problems.
-def test_defined_url(single_record_xml, single_record_response, kielipankki_api_url):
+
+def test_send_data_to_metax_no_records_post(shared_request_mocker, metax_base_url):
     """
-    Test that the CLI can fetch records from a specific URL
+    Check that send_data_to_metax does not POST if en empty dictinary is passed to it.
     """
-    runner = CliRunner()
+    record_dict = {}
+    metadata_harvester_cli.send_data_to_metax(record_dict)
+    shared_request_mocker.post(metax_base_url)
 
-    result = runner.invoke(retrieve_metadata_content, ["--url", kielipankki_api_url], input=single_record_xml)
-    print(result)
+    assert shared_request_mocker.call_count == 0
 
-    assert single_record_response["persistent_identifier"] in result.output
-    assert single_record_response["title"]["en"] in result.output
-    assert single_record_response["title"]["fi"] in result.output
-    assert single_record_response["modified"] in result.output
-    assert single_record_response["issued"] in result.output
-    assert single_record_response["description"]["en"] in result.output
-    assert single_record_response["description"]["fi"] in result.output 
 
-def test_default_url(single_record_xml, single_record_response):
+def test_send_data_to_metax_no_records_put(shared_request_mocker, metax_base_url):
     """
-    Test that the CLI can fetch records from the default URL
+    Check that send_data_to_metax does not PUT if an empty dictionary is passed to it.
     """
-    runner = CliRunner()
+    record_dict = {}
+    metadata_harvester_cli.send_data_to_metax(record_dict)
+    shared_request_mocker.put(metax_base_url)
+
+    assert shared_request_mocker.call_count == 0
+
 
-    result = runner.invoke(retrieve_metadata_content, input=single_record_xml)
+def test_main_all_data_harvested(
+    mock_requests_post,
+    mock_metashare_record_not_found_in_datacatalog,
+    single_record_to_dict):
+    """
+    Check that when no successful harvest date is available, all data is fetched from Kielipankki
+    and then sent to Metax.
+    """
+    metadata_harvester_cli.main()
+
+    assert mock_requests_post.call_count == 3
+    assert mock_requests_post.request_history[2].method == "POST"
+    assert mock_requests_post.request_history[2].json()["persistent_identifier"] == list(single_record_to_dict.values())[0]["persistent_identifier"]
+
+
+def test_main_new_records_harvested_since_date(
```

Comment:
And here we could make sure that something that isn't meant to be updated doesn't, and something that is is.